legend-engine-xt-analytics-store-entitlement-api/src/main/java/org/finos/legend/engine/api/analytics/StoreEntitlementAnalytics.java

```diff
@@ -56,50 +62,49 @@ public StoreEntitlementAnalytics(ModelManager modelManager)
     }
 
     @POST
-    @Path("datasetSpecification")
-    @ApiOperation(value = "generate datasetSpecifications")
-    @Consumes({MediaType.APPLICATION_JSON})
+    @Path("surveyDatasets")
+    @ApiOperation(value = "generate dataset specifications")
+    @Consumes({MediaType.APPLICATION_JSON, APPLICATION_ZLIB})
+    @Produces(MediaType.APPLICATION_JSON)
     public Response generateDatasetSpecifications(StoreEntitlementAnalyticsInput input, @ApiParam(hidden = true) @Pac4JProfileManager ProfileManager<CommonProfile> pm)
     {
         MutableList<CommonProfile> profiles = ProfileManagerHelper.extractProfiles(pm);
         List<EntitlementServiceExtension> extensions = EntitlementServiceExtensionLoader.extensions();
-        PureModel pureModel = modelManager.loadModel(input.model, PureClientVersions.production, profiles, null);
+        PureModel pureModel = modelManager.loadModel(input.model, input.clientVersion == null ? PureClientVersions.production : input.clientVersion, profiles, null);
         Mapping mapping = pureModel.getMapping(input.mappingPath);
         Root_meta_pure_runtime_Runtime runtime = pureModel.getRuntime(input.runtimePath);
-        try (Scope scope = GlobalTracer.get().buildSpan("generate datasetSpecifications").startActive(true))
+        try (Scope scope = GlobalTracer.get().buildSpan("generate dataset specifications").startActive(true))
+        {
+            List<DatasetSpecification> datasets = LazyIterate.flatCollect(extensions, extension -> extension.generateDatasetSpecifications(input.query, input.runtimePath, runtime, input.mappingPath, mapping, input.model, pureModel)).toList();
+            return ManageConstantResult.manageResult(profiles, new SurveyDatasetsResult(datasets), EntitlementModelObjectMapperFactory.getNewObjectMapper());
```

Comment:
TODO: create a common private static final `object mapper` instead so we don't need to build it on every API call